mainwindow.cpp

```diff
@@ -11,23 +10,25 @@ using jbrush::Canvas;
 using jbrush::Canvas;
 
 MainWindow::MainWindow(QWidget *parent) :
-    QMainWindow(parent),
-    ui(new Ui::MainWindow)
+    QMainWindow(parent), ui(new Ui::MainWindow), canvas(new jbrush::Canvas)
 {
     setWindowIcon(QIcon(":/resources/icon.png"));
     ui->setupUi(this);
 
-    Canvas *canvas = new Canvas;
-    MainWindow::setCentralWidget(canvas);
-
-    QPoint p1(100,100);
-    QPoint p2(200,200);
-    jbrush::Shape* shape = new jbrush::Rect(p1,p2);
-
-    canvas->addShape(shape);
+    setCentralWidget(canvas);
 }
 
 MainWindow::~MainWindow()
 {
     delete ui;
+    delete canvas;
+}
+
+void MainWindow::on_actionLoad_triggered()
+{
+    // Bring up a panel for the user to select a directory
+    currentFile = QFileDialog::getOpenFileName(canvas, "Open", currentFile, "Jojibrush files (*" + jbrush::jconstants::FILE_EXTENSION + ")");
```

Comment:
Wouldn't ``this`` be better than ``canvas``? We want the dialog to open with respect to ``MainWindow``
```suggestion
    currentFile = QFileDialog::getOpenFileName(this, "Open", currentFile, "Jojibrush files (*" + jbrush::jconstants::FILE_EXTENSION + ")");
```